README.md

```diff
@@ -1621,6 +1622,44 @@ authClient.token.getWithPopup(options)
 });
 ```
 
+#### `token.getWithIDPPopup(options)`
+
+> :exclamation: Read tradeoffs carefully
+
+> :link: web browser only <br>
+> :hourglass: async
+
+Using [External Identity Providers](https://developer.okta.com/docs/concepts/identity-providers/) in conjunction with [`token.getWithPopup`](#tokengetwithpopupoptions) can fail when an external (non-Okta) IDP sets their [`Cross-Origin-Opener-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy) to something other than `unsafe-none` on _any_ document loaded in the authentication flow. This causes the spawned popup window and main browser window to run in isolated `Browser Context Groups` ([BCG](https://developer.mozilla.org/en-US/docs/Glossary/Browsing_context)); this results in the following
+
+1. The popup and main window can no longer communicate via [`window.postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage)
+2. The main window can no longer detect if the popup window is closed
+
+[`token.getWithIDPPopup`](#tokengetwithidppopupoptions) is designed for deployments which require the use of a popup window _and_ rely on external IDPs. This method can authenticate a user regardless of the IDP's [`Cross-Origin-Opener-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy), however it does come with some tradeoffs
+
+##### Usage
+```javascript
+const { promise, cancel } = authClient.token.getWithIDPPopup({
+  redirectUri: 'http://localhost:8080/popup/callback',
+});
+const { tokens } = await promise;
+authClient.tokenManager.setTokens(tokens);
+```
+
+#### Tradeoffs
+1. Since [`window.postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) can no longer be replied on, the popup window's flow needs to end on the same origin as the application requesting tokens. After successful authentication, the popup window will be redirect to the provided `redirectUri`, which needs to call `authClient.handleIDPPopupRedirect()`. In Single Page Apps (with a router), it's recommended to host a dedicated route, but this logic can be perform on the main page as well.
```

Comment:
> can no longer be replied on

typo: can no longer be relied upon

> but this logic can be perform on the

typo: but this logic can be performed on the